.github/workflows/code-scan.yml

```diff
@@ -0,0 +1,76 @@
+name: Code Quality and Security Scanning
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  codeql:
+    name: Analyze (CodeQL)
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+    strategy:
+      matrix:
+        language: [ 'java', 'cpp' ]  # Adjust based on the languages in the repo
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: ${{ matrix.language }}
+
+      - name: Autobuild for Java
+        if: matrix.language == 'java'
+        run: |
+          mvn clean install
+
+      - name: Build for C++
+        if: matrix.language == 'cpp'
+        run: |
+          mkdir build
+          cd build
+          cmake ..
+          make
+
+      - name: Perform CodeQL analysis
+        uses: github/codeql-action/analyze@v2
+
+  lint_java:
```

Comment:
Let's remove linting jobs, as linting alerts are already part of [codacy reports](https://app.codacy.com/gh/thepurpleowl/House-of-Codes/issues/current)

Reply:
removed